src/common/SurgeStorage.cpp

```diff
@@ -28,8 +30,34 @@ float samplerate, samplerate_inv;
 double dsamplerate, dsamplerate_inv;
 double dsamplerate_os, dsamplerate_os_inv;
 
+#if MAC
+#include <CoreFoundation/CoreFoundation.h>
+string getSelfLocation() {
+    char path[PATH_MAX];
+    // TODO: use a build-provided symbol
+    CFStringRef selfName = CFSTR("com.vemberaudio.plugins.surge");
+    CFBundleRef mainBundle = CFBundleGetBundleWithIdentifier(selfName);
+    CFURLRef resourcesURL = CFBundleCopyBundleURL(mainBundle);
+    CFStringRef str = CFURLCopyFileSystemPath( resourcesURL, kCFURLPOSIXPathStyle );
+    CFRelease(resourcesURL);
+    CFStringGetCString( str, path, FILENAME_MAX, kCFStringEncodingASCII );
+    CFRelease(str);
+    string out(path);
+    return out;
+}
+#endif
+
 SurgeStorage::SurgeStorage()
 {
+#if MAC
+    // Quick hack to install all the bundled surge data to user's local ~/Library/...
+    fs::path userSurgeDir(string(getenv("HOME")) + "/Library/Application Support/Surge");
+    if (!fs::is_directory(userSurgeDir)) {
+        fs::create_directories(userSurgeDir);
+        fs::copy_recursive(fs::path(getSelfLocation() + "/Contents/Data"), userSurgeDir);
+    }
```

Comment:
I worry a little bit that we are doing this full recursive copy every time we make a new instance of Surge. If you have 4 or 5 instances running this is quite a bit of file churn.

Perhaps we want to do something like have a version file and copy when that revs?

I'm all for this being in the bundle and auto-installing, but I think doing it each and every time is probably overkill relative to how often it will change.

That said, I'm happy to proceed with this and add an issue and a fixme to make this more parsimonious in the future. Clever change!

Reply:
The fact it's currently just checking for the existence of the data directory definitely could be an issue getting updates with subsequent releases applied, but I'm also not really sure how that's been handled in the face of user editing of that stuff. If users are not supposed to edit the configuration file(s) but just add waveforms, etc., then it would make the most sense to just load it up from the bundle as I originally suggested and have multiple locations to read data from.

Mind you, I'm not at all opposed to an installer and ditching this behavior, I just don't want to force users to run a command line script to get up and running and I am opposed to dumping things on the filesystem for no good reason.

It would be trivial to add a version file somewhere and check that to determine if resources needed to be synchronized. If we moved this to a static initializer (in fact this would be much better than in the `SurgeStorage` class) that happened at bundle load time, it would avoid checking each time you load a new instance. Of course, it's already reading that whole directory each time, anyway, so just checking a version file would be a drop in the bucket comparatively.

Reply:
Oh duh sorry 

Well let’s add a comment or add to an issue to think about how to rev this. Some sum of the contents or what not. Or as you say a proper installer! But apologize for missing that!!

Reply:
@baconpaul agreed, added a comment about it and possible solution to deal with updates in the future